From 50684a580b47a33623d275725e2374edc7e31052 Mon Sep 17 00:00:00 2001
From: Roel Janssen <roel@gnu.org>
Date: Mon, 5 Feb 2018 11:27:14 +0100
Subject: [PATCH 2/2] Adapt Cobalt command-line options.

* lib/HMF/Pipeline/Cobalt.pm: Adjust Cobalt's run function to run for pairs.
* templates/Cobalt.sh.tt: Adjust command-line arguments.
---
 lib/HMF/Pipeline/Cobalt.pm | 40 ++++++++++++++++++----------------------
 templates/Cobalt.sh.tt     |  7 +++++--
 2 files changed, 23 insertions(+), 24 deletions(-)

diff --git a/lib/HMF/Pipeline/Cobalt.pm b/lib/HMF/Pipeline/Cobalt.pm
index 4b09d3d..5554cb5 100644
--- a/lib/perl5/site_perl/5.26.0/HMF/Pipeline/Cobalt.pm
+++ b/lib/perl5/site_perl/5.26.0/HMF/Pipeline/Cobalt.pm
@@ -17,28 +17,24 @@ sub run {
 
     say "\n### SCHEDULING COBALT ANALYSIS ###";
 
-    my @cobalt_jobs;
-    foreach my $sample (keys %{$opt->{SAMPLES}}) {
-        my ($sample_bam, $running_jobs) = sampleBamAndJobs($sample, $opt);
-        my $dirs = createDirs($opt->{OUTPUT_DIR}, cobalt => "cobalt");
-
-        my $job_id = fromTemplate(
-            "Cobalt",
-            $sample,
-            1,
-            qsubJava($opt, "COBALT"),
-            $running_jobs,
-            $dirs,
-            $opt,
-            sample => $sample,
-            sample_bam => $sample_bam,
-        );
-        next unless $job_id;
-
-        push @cobalt_jobs, $job_id;
-    }
-    $opt->{RUNNING_JOBS}->{'cobalt'} = \@cobalt_jobs;
-    return;
+    my ($ref_sample, $tumor_sample, $ref_bam_path, $tumor_bam_path, $joint_name, $running_jobs) = sampleControlBamsAndJobs($opt);
+    my $dirs = createDirs($opt->{OUTPUT_DIR}, cobalt => "cobalt");
+
+    my $job_id = fromTemplate(
+        "Cobalt",
+        undef,
+        1,
+        qsubJava($opt, "COBALT"),
+        $running_jobs,
+        $dirs,
+        $opt,
+        ref_sample => $ref_sample,
+        ref_bam_path => $ref_bam_path,
+        tumor_sample => $tumor_sample,
+        tumor_bam_path => $tumor_bam_path,
+    );
+
+    return $job_id;
 }
 
 1;
\ No newline at end of file
diff --git a/templates/Cobalt.sh.tt b/templates/Cobalt.sh.tt
index c63e4c8..2a07960 100644
--- a/share/hmf-pipeline/templates/Cobalt.sh.tt
+++ b/share/hmf-pipeline/templates/Cobalt.sh.tt
@@ -6,7 +6,10 @@
 
 java -Xmx[% opt.COBALT_MEM %]G \
     -cp "[% opt.COBALT_PATH %]/cobalt.jar" com.hartwig.hmftools.cobalt.CountBamLinesApplication \
-    -input [% sample_bam %] \
-    -sample [% sample %] \
-    -output_dir [% dirs.cobalt %] \
+    -gc_profile "[% opt.GC_PROFILE %]" \
+    -reference "[% ref_sample %]" \
+    -reference_bam "[% ref_bam_path %]" \
+    -tumor "[% tumor_sample %]" \
+    -tumor_bam "[% tumor_bam_path %]" \
+    -output_dir "[% dirs.cobalt %]" \
     -threads [% opt.COBALT_THREADS %]

success
-- 
2.15.0

